### PR TITLE
Use parent for bfs validation

### PR DESCRIPTION
**Pull request by nicelhc13**
Since other variants than direction optimization are using distance labels, CI test would be failed.

### CODE REVIEW

File: scripts/bench_python_cpp_algos.py [hunk 1]
```diff
@@ -36,8 +36,8 @@ def run_bfs(property_graph: PropertyGraph, input_args, source_node_file):
 
     bfs_plan = analytics.BfsPlan.synchronous_direction_opt(15, 18)
 
-    if "road" in input_args["name"]:
-        bfs_plan = analytics.BfsPlan.asynchronous()
+    # if "road" in input_args["name"]:
+    #        bfs_plan = analytics.BfsPlan.asynchronous()
```
Comment (arthurp) [inline scripts/bench_python_cpp_algos.py]: Remove instead of comment.

**Review by arthurp — COMMENTED** [review]
This will be rebased on top of your other parent PR?

**Comment by nicelhc13** [comment]
> This will be rebased on top of your other parent PR?

No, I think that PR is still under discussion and is not clear to me yet. As @roshandathathri  suggested, our focus for now is to make direction-optimization works.

**Comment by arthurp** [comment]
> > This will be rebased on top of your other parent PR?
> 
> No, I think that PR is still under discussion and is not clear to me yet. As @roshandathathri suggested, our focus for now is to make direction-optimization works.

This PR doesn't really make any sense without the other changes. So this will also need to wait until we figure out the plan for BFS and SSSP outputs.

File: libgalois/src/analytics/bfs/bfs.cpp [hunk 2]
```diff
@@ -500,33 +500,82 @@ katana::analytics::Bfs(
...
-  do_all(
-      iterate(graph),
-      BfsImplementation::NotConsistent<BfsNodeDistance, BfsNodeDistance>(
-          &graph, not_consistent));
-
-  if (not_consistent) {
-    return katana::ErrorCode::AssertionFailed;
+  for (uint32_t u : graph) {
+    // TODO(lhc): should be BfsParent
+    //            (keep this since still exist variants using distance as label)
```
Comment (roshandathathri) [inline libgalois/src/analytics/bfs/bfs.cpp]: Change "since still exist" to "since there exists"

File: libgalois/src/analytics/bfs/bfs.cpp [hunk 3]
```diff
@@ -500,33 +500,82 @@ katana::analytics::Bfs(
...
+    uint32_t u_parent = graph.GetData<BfsNodeDistance>(u);
```
Comment (roshandathathri) [inline libgalois/src/analytics/bfs/bfs.cpp]: Why is "BfsNodeDistance" a parent? Rename that to "BfsNodeParent".

**Review by roshandathathri — APPROVED** [review]
LGTM but let's wait for @arthurp's approval (and changes)

File: libgalois/src/analytics/bfs/bfs.cpp (hunk 3 above)
Comment (nicelhc13) [reply libgalois/src/analytics/bfs/bfs.cpp]: This was b/c we didn't decide parent vs distance. Other variants have used distance. Now it is good to change to parent.

**Review by arthurp — CHANGES_REQUESTED** [review]
* Please disable the other algorithms in this PR as well. You can just comment out the constructors on the plan (in C++ and Python) and change the default to the algorithm we are leaving.
* Update the documentation to match what we are really computing.

File: libgalois/include/katana/analytics/bfs/bfs.h [hunk 4]
```diff
@@ -95,9 +95,7 @@ struct KATANA_EXPORT BfsStatistics {
   /// The number of nodes reachable from the source node.
   uint64_t n_reached_nodes;
   /// The maximum distance across all nodes.
-  uint32_t max_distance;
-  /// The average distances on visited nodes from the source node.
-  double average_visited_distance;
+  uint32_t max_parent;
```
Comment (arthurp) [inline libgalois/include/katana/analytics/bfs/bfs.h]: What is the `max_parent` and why would I care? Maybe just drop this entirely and we can put meaningful stats back in a bit later. 

Also docs need updating here.

Comment (nicelhc13) [reply libgalois/include/katana/analytics/bfs/bfs.h]: Hehe yeah, let me remove this! Yeah I am gonna update documents.

Comment (roshandathathri) [reply libgalois/include/katana/analytics/bfs/bfs.h]: Agree. Not useful.

**Review by arthurp — CHANGES_REQUESTED** [review]
What happens if I call with an unsupported plan? For now we could just return `katana::ErrorCode::NotImplemented`. Currently it looks like it would run the algorithm which will do the wrong thing.

File: python/katana/analytics/_bfs.pyx [hunk 5]
```diff
@@ -190,7 +190,7 @@ def bfs_assert_valid(PropertyGraph pg, str property_name):
     output_property_name_bytes = bytes(property_name, "utf-8")
     output_property_name_cstr = <string>output_property_name_bytes
     with nogil:
-        handle_result_assert(BfsAssertValid(pg.underlying_property_graph(), output_property_name_cstr))
+        handle_result_assert(BfsAssertValid(pg.underlying_property_graph(), start_node, output_property_name_cstr))
```
Comment (arthurp) [inline python/katana/analytics/_bfs.pyx:191]: start_node isn't an argument to this function at the Cython level. You will need to add it above. And add it in the test cases.

Comment (arthurp) [reply python/katana/analytics/_bfs.pyx:191]: I'm fixing this. Commit in coming.

**Review by arthurp — APPROVED** [review]
My issues are addressed. I'm gonna be AFK for a while. So I'm gonna approve to avoid blocking anything.

**Review by roshandathathri — APPROVED** [review]
LGTM too

**Review by roshandathathri — APPROVED** [review]
LGTM

**Pull request merged**